x-pack/plugins/enterprise_search/server/routes/app_search/app_search_request_handler.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import fetch from 'node-fetch';
+import querystring from 'querystring';
+import { Logger } from 'src/core/server';
+import { ConfigType } from '../../index';
+
+interface IAppSearchRequestParams {
+  config: ConfigType;
+  log: Logger;
+  path: string;
+  hasValidData: (body) => boolean;
+}
+
+/**
+ * This function creates a handler, in a standard way, for handling App Search
+ * API requests.
+ *
+ * This handler assumes that it will essentially just proxy the App Search API
+ * request, so the request body and request parameters, and body are simply
+ * passed through.
+ */
+export function createAppSearchRequestHandler({
+  config,
+  log,
+  path,
+  hasValidData,
+}: IAppSearchRequestParams) {
+  return async (_context, request, response) => {
+    try {
+      const enterpriseSearchUrl = config.host as string;
+      const params = querystring.stringify(request.query);
+      const url = `${encodeURI(enterpriseSearchUrl)}${path}?${params}`;
+
+      const apiResponse = await fetch(url, {
+        headers: { Authorization: request.headers.authorization as string },
+      });
+
+      const body = await apiResponse.json();
+
+      if (hasValidData(body)) {
+        return response.ok({ body });
+      } else {
+        // Either a completely incorrect Enterprise Search host URL was configured, or App Search is returning bad data
+        throw new Error(`Invalid data received from App Search: ${JSON.stringify(body)}`);
+      }
+    } catch (e) {
+      log.error(`Cannot connect to App Search: ${e.toString()}`);
+      if (e instanceof Error) log.debug(e.stack as string);
+
+      return response.notFound({ body: 'cannot-connect' });
```

Comment:
I've actually been thinking of was thinking of changing this to a 500 instead of 404 error, but it's weird because technically it's not the internal _Kibana_ server having issues, it's the "external"(?) Enterprise Search server having issues... I dunno, what do you think? Is there a better status code we could use? 🤔

Reply:
Oh!! Maybe [502 Bad Gateway](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/502) - that sounds pretty close to what we want, right?

Reply:
Just tried this out in a separate PR and it works beautifully!

```suggestion
      return response.customError({
        statusCode: 502,
        body: 'Error connecting or fetching data from Enterprise Search',
      });
```

What do you think?